### PR TITLE
Introduce Stripe Payments

### PR DESCRIPTION
**Pull request by dmbch**
This PR introduces basic stripe integration. It is currently running in test mode - feel free to offset using Stripe's documented [cards](https://stripe.com/docs/testing#cards).

### CODE REVIEW

File: hooks/usePayment.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { useState } from 'react';
+import { loadStripe } from '@stripe/stripe-js';
+
+const stripePromise = loadStripe(process.env.STRIPE_PUBLIC_KEY);
+
+const getPaymentURL = (e = 1, n) =>
+  `/api/payment?e=${Number(e)}${n ? `&n=${Number(n)}` : ''}`;
```
Comment (reasn) [inline hooks/usePayment.js:7]: Personal preference: I find this more readable
```js
'/api/payment?e='
  + Number(e)
  + n ? '&n=' + Number(n) : '';
```

File: content/thanks.md [hunk 2]
```diff
@@ -0,0 +1 @@
+**Thanks for all the fish!! 🐟**
```
Comment (reasn) [inline content/thanks.md:1]: Personal preference: Thank you for the fish!

File: components/PaymentButton.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { usePayment, useTrip } from '../hooks';
```
Comment (reasn) [inline components/PaymentButton.js:1]: magic

File: pages/api/payment.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+export default async (req, res) => {
+  const { e = 1, n = 1 } = req.query;
+  const { referer } = req.headers;
+  const { id: sessionId } = await stripe.checkout.sessions.create({
+    submit_type: 'donate',
+    payment_method_types: ['card'],
+    line_items: [
+      {
+        name: 'Decarbonization Fee',
```
Comment (reasn) [inline pages/api/payment.js:21]: Can we have start and destination airports in there?

File: content/thanks.md (hunk 2 above)
Comment (dmbch) [reply content/thanks.md:1]: Bbbut this is the [official version](https://en.wikipedia.org/wiki/So_Long,_and_Thanks_for_All_the_Fish)!

File: components/PaymentButton.js (hunk 3 above)
Comment (dmbch) [reply components/PaymentButton.js:1]: Yup, I used to feel like that before I actually started using hooks; Clarke's third law remains in full effect...

File: hooks/usePayment.js (hunk 1 above)
Comment (dmbch) [reply hooks/usePayment.js]: Changed it to make it use a more proper API (`URLSearchParams`).

**Pull request merged**